### PR TITLE
hooks: converted AppBarSearchContainer.js to hooks

### PR DESCRIPTION
**Pull request by jakegleason**
- [x] Identified the issue which this PR solves.
- [x] Read the [**CONTRIBUTING**](https://github.com/target/goalert/blob/master/CONTRIBUTING.md) document.
- [x] Code builds clean without any errors or warnings.
- [x] Added appropriate tests for any new functionality.
- [x] All new and existing tests passed.
- [x] Added comments in the code, where necessary.
- [x] Ran `make check` to catch common errors. Fixed any that came up.

**Description:**

converted AppBarSearchContainer.js to hooks

### CODE REVIEW

File: web/src/app/util/AppBarSearchContainer.js [hunk 1]
```diff
@@ -125,3 +116,11 @@ export default function AppBarSearchContainer(props) {
     </SearchContext.Consumer>
   )
 }
+
+SearchProvider.propTypes = {
+  actions: null,
+}
```
Comment (Forfold) [inline web/src/app/util/AppBarSearchContainer.js:122]: I don't think this is needed, as `actions` is used in `state` as opposed to coming from a `prop`

File: web/src/app/util/AppBarSearchContainer.js [hunk 2]
```diff
@@ -14,94 +14,85 @@ export function SearchContainer() {
...
-      >
-        {this.props.children}
-      </SearchContext.Provider>
-    )
-  }
+  return (
+    <SearchContext.Provider
+      value={{
+        actions: actions,
+        setActions: props.setActions,
```
Comment (Forfold) [inline web/src/app/util/AppBarSearchContainer.js:67]:
```suggestion
        setActions: handleSetActions,
```

File: web/src/app/util/AppBarSearchContainer.js (hunk 1 above)
Comment (jakegleason) [reply web/src/app/util/AppBarSearchContainer.js:122]: Removed actions.

File: web/src/app/util/AppBarSearchContainer.js (hunk 2 above)
Comment (jakegleason) [reply web/src/app/util/AppBarSearchContainer.js:67]: Changed `props.setActions` to `handleSetActions`. Tested by dragging browser to a smaller width.

**Comment by Forfold** [comment]
Closing per our conversation with @mastercactapus that we may need to restructure our render tree a bit to achieve the desired results using functional components in the way we want.

**Pull request closed without merging**